src/lib.rs

```diff
@@ -1,86 +1,12 @@
-//! # Hades252
-//!
-//! Implementation of Hades252 over the Ristretto Scalar field.
-//!
-//! *Unstable* : No guarantees can be made regarding the API stability.
-//!
-//! ## Parameters
-//!
-//! - p = `2^252 + 27742317777372353535851937790883648493`
-//!
-//! - Security level is 126 bits
-//!
-//! - width = 9
-//!
-//! - Number of full rounds = 8 . There are four full rounds at the beginning and four full rounds at the end,
-//! where each full round has `WIDTH` quintic S-Boxes.
-//!
-//! - Merkle tree arity : 8 to 1
-//!
-//! - Number of partial rounds = 59, where each partial round has one quintic S-Box and (width-1) identity functions.
-//!
-//! - Number of round constants = 960
-//!
-//! ## Example outside R1CS
-//! ```
-//! use hades252::scalar::hash;
-//! use curve25519_dalek::scalar::Scalar;
-//!
-//! // Generate the inputs that will be hashed.
-//! // Since the `WIDTH` is set to `9` we can use `8`
-//! // inputs as much.
-//! // If we use less, the function will take care of padding them
-//! // properly.
-//! let input = [Scalar::one(); 7];
-//!
-//! // Hash the data and get the resulting `Scalar`.
-//! let digest_res = hash(&input).unwrap();
-//! ```
-//!
-//! ## Example inside R1CS
-//! ```
-//! use hades252::linear_combination::hash;
-//! use curve25519_dalek::scalar::Scalar;
-//! use bulletproofs::r1cs::{LinearCombination, Verifier};
-//! use merlin::Transcript;
-//!
-//! // Generate the inputs that will be hashed.
-//! // Since the `WIDTH` is set to `9` we can use `8`
-//! // inputs as much.
-//! // If we use less, the function will take care of padding them
-//! // propperly.
-//! let inp = LinearCombination::from(Scalar::one());
-//! let mut lc_one: Vec<LinearCombination> = vec![];
-//! for _ in 0..7 {
-//!     lc_one.push(inp.clone());
-//! };
-//!
-//! // Generate the Verifier (Prover is not generated)
-//! // since this is just a demo.
-//! let mut verifier_transcript = Transcript::new(b"");
-//! let mut verifier = Verifier::new(&mut verifier_transcript);
-//!
-//! // Hash the data and get the resulting `LinearCombination`.
-//! let digest_res = hash(&mut verifier, &lc_one).unwrap();
-//! ```
-//!
-//! ## Deviations
-//!
-//! - Round constants for the full rounds are generated following: https://extgit.iaik.tugraz.at/krypto/hadesmimc/blob/master/code/calc_round_numbers.py
-//! They are then mapped onto `Scalar`s in the Ristretto scalar field.
-//!
-//! - The MDS matrix is a cauchy matrix, the method used to generate it, is noted in section "Concrete Instantiations Poseidon and Starkad"
-//!
-//! ## Reference
-//!
-//! https://eprint.iacr.org/2019/458.pdf
-
+#![feature(trait_alias)]
+#![feature(external_doc)]
+#![deny(missing_docs)]
+#![doc(include = "../README.md")]
```

Comment:
Invalid path reference. Maybe rename the root readme to uppercase?

https://github.com/dusk-network/Hades252/pull/36/commits/6d3e47a9c89888d28e0c309043cd9395acfa5966

Reply:
Yeah, I notice that. I modified that when I committed, but apparently git didn't register the difference between uppercase and lowercase. I had to delete the file and re-add.

P.S.
For future reference: **never** create a branch for a review; **always** add your suggestion directly in the comment.

That for at least two reasons:

1. We don't keep branches alive forever, usually they're kept only until we merge it and become obsolete: therefore if we delete a branch that you create for a review, the review will miss the code suggestion that you made, and that is _very bad_.

2. We don't want to work twice: the reviewer should *never* rewrite the original code submitted for review, but only suggest – using code snippet if needed, see the 1st point –  so that the original author can write it.

2a. As side note of the 2nd point, there is a big difference between *improvement* and a *fix* in a review. The reviewer should *always* point out where the code is not correct, and that should be taken in account in the same PR; however if the code itself is correct and there is something to improve / change, this might be done or discuss as follow up PR (of course the reviewer can point it out in the current PR); so that the code can land – and don't be a blocker for others to use it.